packages/python-google-compute-engine/google_compute_engine/networking/network_daemon.py

```diff
@@ -98,10 +103,13 @@ def HandleNetworkInterfaces(self, result):
       self.network_setup.EnableNetworkInterfaces(
           [interface.name for interface in network_interfaces[1:]])
 
-    for interface in network_interfaces:
-      if self.ip_forwarding_enabled:
-        self.ip_forwarding.HandleForwardedIps(
-            interface.name, interface.forwarded_ips, interface.ip)
+    if self.ip_forwarding_enabled:
+      for interface in network_interfaces:
+        self.ip_forwarding.HandleForwardedIps(interface.name,
+                                              interface.forwarded_ips,
+                                              interface.ip)
+    if socket.gethostname() != result['hostname']:
```

Comment:
Do we know for certain that we will always match the FQDN returned and now get into a loop where it will not match ever? I could imagine this returning values that may never match depending on the distro and/or what a user may have changed. I by the way have no evidence- just asking ;)